File: Week 1/id_455/LeetCode_26_455.c
/*
 * @lc app=leetcode.cn id=26 lang=c
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start


int removeDuplicates(int* nums, int numsSize){
    if(numsSize == 0) return 0;
    int i=0;
    for(int j = 1; j < numsSize; j++){
        if(nums[i] != nums[j]){
            nums[++i]=nums[j];
        }
    }
    return i+1;
}


// @lc code=end

File: Week 1/id_455/LeetCode_26_455.py
#
# @lc app=leetcode.cn id=26 lang=python3
#
# [26] 删除排序数组中的重复项
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        pre,cur = 0,1
        while cur < len(nums):
            if nums[cur] == nums[pre]:
                nums.pop(cur)
            else:
                pre,cur = pre+1,cur+1
        return len(nums)

# @lc code=end

